README.md

```diff
@@ -60,77 +62,60 @@ Alright, let's get started!
+Our goal is to build a scalable system that can monitor multiple cryptocurrency pairs across different workers in real time. By crafting an asynchronous function to connect to the Coinbase Pro WebSocket, we facilitate streaming of cryptocurrency data into our dataflow. This process involves the `websockets` Python library for WebSocket connections and `bytewax` for dataflow integration.
 
-https://github.com/bytewax/crypto-orderbook-app/blob/049229fa01184127658d40d3b47638232038371b/dataflow.py#L9-L27
+The function `_ws_agen` inputs cryptocurrency pair identifiers (e.g., `["BTC-USD", "ETH-USD"]`), establishing a connection to Coinbase Pro's WebSocket. It subscribes to the `level2_batch` channel for live order book updates, sending a JSON subscription message and awaiting a confirmation response with `ws.recv()`.
 
-Now that we have our websocket based data generator built, we will our dataflow. Since we're using a `StatelessSource`, we'll create a `DynamicInput` subclass called `CoinbaseInput`. An instance of `CoinbaseInput` will be constructed on each worker. In the `build` method, we receive information about which worker we are: `worker_index` and the total number of workers (`worker_count`). We have added some custom code in order to distribute the currency pairs with the logic. It should be noted that if you run more than one worker with only one currency pair, the other workers will not be used.
+https://github.com/bytewax/crypto-orderbook-app/blob/b61e9101cbf11dcb05209ffc6c585148fddea312/dataflow.py#L15-L31
 
-https://github.com/bytewax/crypto-orderbook-app/blob/049229fa01184127658d40d3b47638232038371b/dataflow.py#L30-L40
+To efficiently process and manage this data, we implement the `CoinbasePartition` class, extending Bytewax's `StatefulSourcePartition`. 
 
-## Constructing The Dataflow
+This enables:
 
-Before we get to the exciting part of our order book dataflow, we need to create our Dataflow object and prep the data. We'll start with creating an empty `Dataflow` object and add the `CoinbaseSource` we created above.
+* State Management: Maintaining state across data batches.
+* Parallel Processing: Facilitating data processing across multiple workers for each cryptocurrency pair, enhancing scalability.
+* Taking snapshots for Fault Tolerance: Allowing state capture for reliability.
```

Comment:
Although technically the stateful partitions do allow for progress to match state for recovery purposes. Since the websocket isn't replayable, it won't work exactly as advertised. But the websocket helps us out by providing the current orderbook at the beginning of the stream when you subscribe. I am not sure if this is worth mentioning here or not.

Reply:
Addressed here

https://github.com/bytewax/crypto-orderbook-app/pull/2/commits/989e563c679b36cec157229742b0ee3c473e8eec

Replaced the list with your comment - were you thinking of adding the comment after the list?

Reply:
That looks good how it is now.